src/test/java/org/cryptomator/common/mount/MountWithinParentUtilTest.java

```diff
@@ -0,0 +1,210 @@
+package org.cryptomator.common.mount;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryNotEmptyException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+
+import static java.nio.file.LinkOption.NOFOLLOW_LINKS;
+import static org.cryptomator.common.mount.MountWithinParentUtil.cleanup;
+import static org.cryptomator.common.mount.MountWithinParentUtil.getHideaway;
+import static org.cryptomator.common.mount.MountWithinParentUtil.prepareParentNoMountPoint;
+import static org.cryptomator.common.mount.MountWithinParentUtil.removeResidualHideaway;
+import static org.cryptomator.common.mount.MountWithinParentUtil.removeResidualJunction;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+class MountWithinParentUtilTest {
+
+	@Test
+	void testPrepareNeitherExist(@TempDir Path parentDir) {
+		assertThrows(MountPointNotExistingException.class, () -> {
+			prepareParentNoMountPoint(parentDir.resolve("mount"));
+		});
+	}
+
+	@Test
+	void testPrepareOnlyHideawayFileExists(@TempDir Path parentDir) throws IOException {
+		var mount = parentDir.resolve("mount");
+		var hideaway = getHideaway(mount);
+		Files.createFile(hideaway);
+
+		assertThrows(HideawayNotDirectoryException.class, () -> {
+			prepareParentNoMountPoint(mount);
+		});
+	}
+
+	@Test
+	void testPrepareOnlyHideawayDirExists(@TempDir Path parentDir) throws IOException {
+		var mount = parentDir.resolve("mount");
+		var hideaway = getHideaway(mount);
+		Files.createDirectory(hideaway);
+		assertFalse(isHidden(hideaway));
+
+		prepareParentNoMountPoint(mount);
+
+		assumeTrue(isHidden(hideaway));
```

Comment:
That is... not what i meant. You should assume, that the OS is Windows, and if so, assert that the file is hidden.

Reply:
Sorry, my bad! That makes sense :D

Reply:
Resolved by 59d89faf3858c4cc89a7fc08d3ebf2427c5a629f.